olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.cpp

```diff
@@ -599,47 +636,37 @@ void NetworkWinHttp::RequestCallback(HINTERNET, DWORD_PTR context, DWORD status,
       }
     }
 
-    if (callback && handle->http_request) {
-      DWORD wide_len;
-      WinHttpQueryHeaders(handle->http_request, WINHTTP_QUERY_RAW_HEADERS,
-                          WINHTTP_HEADER_NAME_BY_INDEX,
-                          WINHTTP_NO_OUTPUT_BUFFER, &wide_len,
-                          WINHTTP_NO_HEADER_INDEX);
-      if (GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
-        DWORD len = wide_len / sizeof(WCHAR);
-        auto wide_buffer = std::make_unique<WCHAR[]>(len);
-        if (WinHttpQueryHeaders(handle->http_request, WINHTTP_QUERY_RAW_HEADERS,
-                                WINHTTP_HEADER_NAME_BY_INDEX, wide_buffer.get(),
-                                &wide_len, WINHTTP_NO_HEADER_INDEX)) {
-          // Text should be converted back from the wide char to properly handle
-          // UTF-8.
-          auto buffer = std::make_unique<char[]>(len);
-          const int convertResult = WideCharToMultiByte(
-              CP_ACP, 0, wide_buffer.get(), len, buffer.get(), len, 0, nullptr);
-          assert(convertResult == static_cast<int>(len));
-
-          DWORD start = 0, index = 0;
-          while (index < len) {
-            if (buffer[index] == 0) {
-              if (start != index) {
-                std::string entry(&buffer[start], index - start);
-                size_t pos = entry.find(':');
-                if (pos != std::string::npos && pos + 2 < entry.size()) {
-                  std::string key(entry.begin(), entry.begin() + pos);
-                  std::string value(entry.begin() + pos + 2, entry.end());
-                  callback(key, value);
-                }
+    DWORD headers_size = 0;
+
+    if (handle->http_request) {
+      if (callback) {
+        auto headers = QueryHeaders(handle->http_request);
+        DWORD start = 0, index = 0;
+        while (index < headers.size()) {
+          if (headers[index] == 0) {
+            if (start != index) {
+              std::string entry(&headers[start], index - start);
+              size_t pos = entry.find(':');
+              if (pos != std::string::npos && pos + 2 < entry.size()) {
+                std::string key(entry.begin(), entry.begin() + pos);
+                std::string value(entry.begin() + pos + 2, entry.end());
+                callback(key, value);
               }
-              index++;
-              start = index;
-            } else {
-              index++;
             }
+            index++;
+            start = index;
+          } else {
+            index++;
           }
         }
+        headers_size += headers.size();
+      } else {
+        headers_size += QueryHeadersSize(handle->http_request);
```

Comment:
If I understand correctly, QueryHeadersSize returns length  of the request, not size in bytes, so probably need to multiply on `sizeof(WCHAR)`

Reply:
It's intentionally since winapi converts the received data to wchar